### PR TITLE
feat: add support for repeatable links in models 

### PR DESCRIPTION
**Pull request by levimykel**
**Resolves**: [Repeatable links project](https://www.notion.so/prismic/Repeatable-Links-Work-1213bee086698021a7f9d5f20e20a47b)

### Description

The first of 3 PRs to support an array of links as well as single links. It would allow devs to not manually wrap their link in a Group. It would also allow editors to have a better experience with a dedicated UI for multiple links.

This PR handles the model changes. The model adds a new `repeat` property to define if the link field is an array or not.

```JSON
{
  "example_link": {
    "type": "Link",
    "config": {
      "label": "My Button Link",
      "placeholder": "My Button Link Placeholder",
      "repeat": false, // true to activate repetition
      "select": null,
      "allowTargetBlank": false,
      "allowText": true
    }
  }
}
```

Other PRS:
- 2 - [Add support for repeatable link values](https://github.com/prismicio/prismic-client/pull/358)
- 3 - [Support repeatable links in helpers & migration](https://github.com/prismicio/prismic-client/pull/359)

### Checklist

- [x] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: src/types/model/contentRelationship.ts [hunk 1]
```diff
@@ -19,5 +19,6 @@ export interface CustomTypeModelContentRelationshipField<
 		customtypes?: readonly CustomTypeIDs[]
 		tags?: readonly Tags[]
 		allowText?: boolean
+		repeat?: boolean
```
Comment (angeloashmore) [inline src/types/model/contentRelationship.ts:22]: ❓ Are we allowing content relationships to be repeatable? I thought we weren't, but correct me if I'm wrong.

File: src/types/model/linkToMedia.ts [hunk 2]
```diff
@@ -14,5 +14,6 @@ export interface CustomTypeModelLinkToMediaField {
 		placeholder?: string
 		select: typeof CustomTypeModelLinkSelectType.Media
 		allowText?: boolean
+		repeat?: boolean
```
Comment (angeloashmore) [inline src/types/model/linkToMedia.ts:17]: ❓ Are we allowing link to media fields to be repeatable? I thought we weren't, but correct me if I'm wrong.

**Review by angeloashmore — COMMENTED** [review]
Content relationship and link to media fields are included in this PR, but I thought we were only enhancing the link field. If we're allowing it on all link-link fields, let me know and I'll approve it as is.

Everything else looks good. 👍

**Comment by dani-mp** [comment]
@angeloashmore changes applied

**Pull request merged**